client/extensions/woocommerce/app/store-stats/utils.js

```diff
@@ -143,5 +173,60 @@ export function formatValue( value, format, code ) {
  */
 export function getDelta( deltas, selectedDate, stat ) {
 	const selectedDeltas = find( deltas, item => item.period === selectedDate );
-	return selectedDeltas[ stat ];
+	return ( selectedDeltas && selectedDeltas[ stat ] ) || [];
+}
+
+/**
+ * Given a date, an array of data, and a stat, return a delta object for the specific stat.
+ *
+ * @param {array} data - an array of API data, must contain at least 3 rows
+ * @param {string} selectedDate - string of date in 'YYYY-MM-DD'
+ * @param {string} stat - string of stat to be referenced
+ * @param {string} unit - unit/period format for the data provided
+ * @return {object} - Object containing data from calculateDelta
+ */
+export function getDeltaFromData( data, selectedDate, stat, unit ) {
+	if ( ! data || ! Array.isArray( data ) || data.length < 3 ) {
+		return {};
+	}
+	let delta = {};
+	let previousItem = false;
+
+	forEach( data, function( item ) {
+		if ( previousItem ) {
+			if ( item.period === selectedDate ) {
+				delta = calculateDelta( item, previousItem, stat, unit );
+			}
+			previousItem = item;
+		} else {
+			previousItem = item;
+		}
+	} );
+
+	return delta;
+}
+
+/**
+ * Given visitor data and order data, get a list conversion rate by period.
+ *
+ * @param {array} visitorData - an array of API data from the 'visits' stat endpoint
+ * @param {array} orderData -  an array of API data from the orders endpoint
+ * @param {string} unit - unit/period format for the data provided
+ * @return {object} - Object containing data from calculateDelta
+ */
+export function getConversionRateData( visitorData, orderData, unit ) {
+	return visitorData.map( visitorRow => {
+		const datePeriod = getEndPeriod( visitorRow.period, unit );
+		const unitPeriod = getUnitPeriod( visitorRow.period, unit );
+		const index = findIndex( orderData, d => d.period === datePeriod );
+		const orders = orderData[ index ] && orderData[ index ].orders;
+
+		if ( visitorRow.visitors > 0 && orderData[ index ] ) {
+			return {
+				period: unitPeriod,
+				conversionRate: round( orders / visitorRow.visitors * 100, 2 ),
```

Comment:
Quick one: are these uniques? If 1 customer makes 2 orders in the period, is that ( 2 orders / 1 visitor ) or ( 2 orders / 2 visitors ) OR ( 1 order / 1 visitor )?

Reply:
@greenafrican 

Visitors is unique. Orders is not. So that would be `( 2 orders / 1 visitor )`.

> then perhaps both stats should be uniques?

I don't think we have the data for unique orders as far as I can tell. Do you know of something?

When searching and doing some research, it looked like orders/actions divided by unique visitors works for calculating your conversion rate, as long as you are being consistent in your other calculations. We could have a help icon that shows what the calculation is?

--

It seems like our options are `orders / visitors` or `orders / unique visitors`

Reply:
I took a look to see if we could made add a new endpoint but yeah, I don't think we have the synced data to do unique orders at this time. See comment #comment-1059 on p90Yrv-Cl-p2. Anything we could use to do unique orders will no longer be synced. I also don't think we would get accurate uniques for guest orders this way. 

I would say we keep this calculation as is (with a little indication what the calculation is) and maybe long term we could add some tracking to Jetpack that helps us get a real uniques order count?